internal/test/e2e/run.go

```diff
@@ -482,3 +468,43 @@ func logTestGroups(logger logr.Logger, instancesConf []instanceRunConf) {
 	}
 	logger.V(1).Info("Running tests in parallel", "testsGroups", testGroups)
 }
+
+func getHardwarePool(storageBucket string) ([]*api.Hardware, error) {
+	awsSession, err := session.NewSession()
+	if err != nil {
+		return nil, fmt.Errorf("creating aws session for tests: %v", err)
+	}
+	err = s3.DownloadToDisk(awsSession, os.Getenv(tinkerbellHardwareS3FileKeyEnvVar), storageBucket, e2eHardwareCsvFilePath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to download tinkerbell hardware csv: %v", err)
+	}
+
+	hardware, err := api.NewHardwareSliceFromFile(e2eHardwareCsvFilePath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get Tinkerbell hardware: %v", err)
+	}
+	return hardware, nil
+}
+
+func reserveTinkerbellHardware(conf instanceRunConf, invQueue *hardwareQueue) error {
+	reservedTinkerbellHardware, err := invQueue.reserveHardware(conf.hardwareCount)
+	if err != nil {
+		return fmt.Errorf("TestRunner timed out waiting for hardware - test instanceId: %v", conf.instanceId)
```

Comment:
What's `TestRunner`? That might be context this function doesn't have.

```suggestion
		return fmt.Errorf("timed out waiting for hardware: test_instance_id=%v", conf.instanceId)
```

Reply:
Test runner is the `instanceRunConf` that is being passed as the first parameter to this function. So the context is there but I think it need not be included in the error message.!